codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -86,30 +86,43 @@ As one dynamic contract register catches up to the fetched blocknumebr of the ne
 merge itself into the next register and combine queries/addresses and queues until fully caught
 up to the root. 
 */
-type fetchStateData = {
+type registerData = {
   latestFetchedBlock: blockNumberAndTimestamp,
   contractAddressMapping: ContractAddressingMap.mapping,
   //Events ordered from latest to earliest
   fetchedEventQueue: array<Types.eventBatchQueueItem>,
   //Used to prune dynamic contract registrations in the event
   //of a rollback.
   dynamicContracts: DynamicContractsMap.t,
-  isFetchingAtHead: bool,
   firstEventBlockNumber: option<int>,
 }
 
-type rec t = {
-  registerType: register,
-  ...fetchStateData,
+type rec register = {
+  registerType: registerType,
+  ...registerData,
+}
+and registerType =
+  | RootRegister({endBlock: option<int>})
+  | DynamicContractRegister({id: EventUtils.eventIndex, nextRegister: register})
+
+type dynamicContractRegistration = {
+  registeringEventBlockNumber: int,
+  registeringEventLogIndex: int,
+  registeringEventChain: ChainMap.Chain.t,
+  dynamicContracts: array<TablesStatic.DynamicContractRegistry.t>,
+}
+type t = {
+  baseRegister: register,
+  pendingDynamicContracts: array<dynamicContractRegistration>,
+  isFetchingAtHead: bool,
```

Comment:
This is the new `FetchState.t`, it holds the baseRegister, an array of pending dynamic contracts that are to be lazily added and the "isFetchingAtHead" state is moved from each register to a single value here.